nuxeo-distribution/nuxeo-launcher/src/test/java/org/nuxeo/launcher/TestNuxeoLauncher.java

```diff
@@ -174,47 +174,24 @@ public void testRegisterTrialIsValidCommand() throws Exception {
     }
 
     /**
-     * A valid command of {@code register-trial} contains 0 or 5 additional arguments:
-     *
-     * <pre>
-     * <code>
-     * register-trial
-     * register-trial firstName lastName email company project
-     * </code>
-     * </pre>
+     * Command "register-trial" is deprecated since 9.3.
      * <p>
      * Set timeout to 1 second, {@code timeout = 1000}, to prevent this test takes too long time to finish.
      * The only case it can happen is when Nuxeo Launcher waits for user
      * value from standard input (stdin). It means somebody has changed
      * the implementation of #registerTrial.
      */
     @Test(timeout = 1000) // 1s. Explanation in Javadoc.
-    public void testRegisterTrialWithWrongNumberArgs() throws Exception {
-        String[][] argsToTest = new String[][] {
-                {"register-trial"}, // OK
-                {"register-trial", "first"},
-                {"register-trial", "first", "last"},
-                {"register-trial", "first", "last", "email"},
-                {"register-trial", "first", "last", "email", "company"},
-                {"register-trial", "first", "last", "email", "company", "project"}, // OK
-                {"register-trial", "first", "last", "email", "company", "project", "tooMany"}
-        };
-
-        for (String[] args : argsToTest) {
-            NuxeoLauncher launcher = NuxeoLauncher.createLauncher(args);
-            String cmdStr = "Command " + Arrays.toString(args);
-
-            // Skip assertions for valid argument length
-            if (args.length == 1 || args.length == 6) {
-                continue;
-            }
-
-            try {
-                launcher.registerTrial();
-                fail(cmdStr + " did not raise exception.");
-            } catch (ConfigurationException e) {
-                assertEquals(cmdStr + ": " + e.getMessage(), "Wrong number of arguments.", e.getMessage());
-            }
+    public void testRegisterTrialIsDeprecated() throws Exception {
+        String[] cmd = { "unused" };
+        NuxeoLauncher launcher = NuxeoLauncher.createLauncher(cmd);
+        try {
+            launcher.registerTrial();
+            fail("Command \"register-trial\" did not raise exception.");
+        } catch (UnsupportedOperationException e) {
+            String msg = "This command is deprecated. To register for a free 30 day trial on Nuxeo Online Services,"
+                    + " please visit https://connect.nuxeo.com/register";
+            assertEquals(msg, e.getMessage());
         }
     }
```

Comment:
An alternative is 
```    @Rule
    public ExpectedException thrown = ExpectedException.none();

    @Test
    public void testRegisterTrialIsDeprecated() throws Exception {
        thrown.expect(UnsupportedOperationException.class);
        thrown.expectMessage("deprecated");
        thrown.expectMessage("https://connect.nuxeo.com/register");
        NuxeoLauncher launcher = NuxeoLauncher.createLauncher(new String[] { "register-trial" });
        launcher.registerTrial();
    }
```

Reply:
Yes but the `ExpectedException` has to be a `@Rule`.